### ISSUE TITLE
Cache for RegExp to improve performance

### ISSUE THREAD

**Issue by ikkisoft** [ikkisoft/SerialKiller#4]
With the current implementation, each regexp lookup involves filesystem access

```
$ inotifywait -mr ./
./ OPEN serialkiller.conf
./ ACCESS serialkiller.conf
./ CLOSE_NOWRITE,CLOSE serialkiller.conf
```

To improve performances, we could create a cache for all blacklists/whitelists - dropped whenever configuration reloads. Potentially, we could also make that cache static and/or thread-local and dependent on the filename, so that two SerialKillers with the same config share the cache

**Comment by ikkisoft** [comment]
Implemented in v0.3

**Comment by ikkisoft** [comment]
I've reverted https://github.com/ikkisoft/SerialKiller/commit/9ef69203de1cd2e5e1b4d86fee9912afe3a6a0d6 as it didn't seem to work properly on the latest version of Commons Configuration. I need to rethink the strategy here.
